src/cargo/util/errors.rs

```diff
@@ -132,6 +132,50 @@ impl fmt::Debug for ProcessError {
 }
 
 // =============================================================================
+// Cargo test errors.
+
+/// Error when testcases fail
+pub struct CargoTestError {
+    pub desc: String,
+    pub exit: Option<ExitStatus>,
+    cause: Option<io::Error>,
+}
+
+impl fmt::Display for CargoTestError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.desc, f)
+    }
+}
+
+impl fmt::Debug for CargoTestError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(self, f)
+    }
+}
+
+impl Error for CargoTestError {
+    fn description(&self) -> &str { &self.desc }
+    #[allow(trivial_casts)]
```

Comment:
I think this may not actually be necessary? (it may be a holdover from another time)